10/umbraco-deploy/handling-cache-refresher-notifications.md

```diff
@@ -0,0 +1,162 @@
+---
+meta.Title: "Handling Cache Refresher Notifications"
+description: "How to respond to deployment events using cache refresher notifications"
+---
+
+# Handling Cache Refresher Notifications
+
+## Background
+
+When you deploy content or other Umbraco data between environments, some notifications that are normally fired in CMS operations are suppressed.
+
+For example, you may handle a `ContentPublishedNotification` to apply some custom logic when a content item is published. This code will run in a normal CMS publish operation. However, when deploying a content item into another environment and triggering it's publishing there, the notification will not be issued. And the custom logic in the notification handler will not run.
+
+This behavior is deliberate and done for performance and reliability reasons. A normal save and publish operation by an editor operates on one item at a time. With deployments, we may have many, and publishing these notifications may lead to at best slow operations, and at worst inconsistent data.
+
+However, what if you do want to run some code on an update, even if this is happening as part of a Deploy operation?
+
+There's an option here using cache refresher notifications. Not all events are suppressed by Umbraco Deploy. Some that are batched up and fired after the deploy operation is completed include those related to refreshing the Umbraco cache.
+
+## Implementing a Cache Refresher Notification
```

Comment:
Have added a reference to that CMS documentation, but I think it does no harm to leave these samples too.  There are some differences - `RefreshByPayload` v `RefreshById` - that are probably worth illustrating.

For the last suggestion, do you think you could write me a paragraph that I can consider, copy and paste in please?  You can't easily work on this PR directly, so that's probably the easiest way. Thanks.